modules/cudaoptflow/include/opencv2/cudaoptflow.hpp

```diff
@@ -65,6 +65,14 @@ namespace cv { namespace cuda {
 // Interface
 //
 
+CV_EXPORTS void buildImagePyramid(const GpuMat& Img, std::vector<GpuMat>& Pyr,
+                                  int maxlevel, Stream& stream);
+
+CV_EXPORTS void sparse(std::vector<GpuMat>& prevPyr, std::vector<GpuMat>& nextPyr,
```

Comment:
> sparse

This API is not needed as "public".

You should reuse existed APIs: `SparseOpticalFlow::calc()`.
You need to add check for parameter type of prev/next images for `STD_VECTOR_CUDA_GPU_MAT` InputArray's kind and provide separate handler for that case.

Reply:
Thanks for the comments. I will remove `sparse` from here. As far as I can figure out, we need only `buildImagePyramid` as public.